pkg/codegen/python/gen.go

```diff
@@ -2440,7 +2440,23 @@ func (mod *modContext) genInitDocstring(w io.Writer, res *schema.Resource, resou
 
 	// If this resource has documentation, write it at the top of the docstring, otherwise use a generic comment.
 	if res.Comment != "" {
-		fmt.Fprintln(b, codegen.FilterExamples(res.Comment, "python"))
+		filteredExamples := codegen.FilterExamples(res.Comment, "python")
+		interpretedRefs := codegen.InterpretPulumiRefs(filteredExamples, func(ref codegen.DocRef) (string, bool) {
+			if ref.Token == "" {
+				return "", false
+			}
+			pyType := strings.ToLower(string(ref.Token.Module())) + "." + string(ref.Token.Name())
+			if ref.Property == "" {
+				return pyType, true
+			}
+			if ref.Token.String() == res.Token {
+				// Property within same type
+				return PyName(ref.Property), true
+			}
+			// Property within another type
+			return strings.ToLower(string(ref.Token.Module())) + "." + string(ref.Token.Name()) + "." + PyName(ref.Property), true
+		})
+		fmt.Fprintln(b, interpretedRefs)
```

Comment:
As I understand it, this will only apply to resource descriptions. Instead of applying `codegen.InterpretPulumiRefs` to individual description fields, maybe we should also add a `func LanguageSpecificDescriptions(s schema.Package, interpretPulumiRefs func(ref codegen.DocRef) (string, bool)) schema.Package` that will map over a schema and transform all description fields.

Reply:
Good question on which fields this applies to. I had assume this is only on the main description fields, but perhaps this should also apply to property descriptions too. 

Not sure on the approach - if it's 'normal' for the codegen to manipulate the original schema passed in. It seems like that could break anywhere that's calling codegen directly, looping over each language as the caller would then need to do deep copies of the schema before calling each codegen to make sure they don't interfere with each other. It would seem that the safer approach would be to call the function at the point of rendering, but it would require each codegen implementation to call in all the right places.

Reply:
I was thinking that `LanguageSpecificDescriptions` would be a copy and mutate operation, not an in-place mutation. I agree we don't want this to mutate for the obvious use case of `for language := languages { generate(schema, language) }`.

I'm fine if the copy is implemented as `b, _ := json.Marshal(schema); _ = json.Unmarshal(b, &newPkg)`.

Reply:
I think a simple `genComment` function in each language should be fine and just wrap each place we interpret a comment string should suffice and be easier to follow in the code.

Reply:
> I had assume this is only on the main description fields, but perhaps this should also apply to property descriptions too.

I'd expect this to work on all description fields.